### PR TITLE
[CHARMAP]Code improvements

### PR DESCRIPTION
**Pull request by peterooch**
## Purpose

Based on the code from nls2txt,
switched the linking from dynamic to static, also simplified the code.

JIRA issue: N/A

## Proposed changes

No functional change, only simplified code.

### CODE REVIEW

File: base/applications/charmap/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,6 @@ list(APPEND SOURCE
 add_rc_deps(charmap.rc ${CMAKE_CURRENT_SOURCE_DIR}/res/charmap.ico)
 add_executable(charmap ${SOURCE} charmap.rc)
 set_module_type(charmap win32gui UNICODE)
-add_importlibs(charmap advapi32 user32 gdi32 comctl32 msvcrt kernel32)
+add_importlibs(charmap advapi32 user32 gdi32 comctl32 msvcrt kernel32 getuname)
```
Comment (ThFabba) [inline base/applications/charmap/CMakeLists.txt]: New exports go to the left, msvcrt/kernel32/ntdll must the the last in the list (if present).

File: base/applications/charmap/charmap.c [hunk 2]
```diff
@@ -268,12 +264,9 @@ UpdateStatusBar(WCHAR wch)
     WCHAR buff[MAX_PATH];
     WCHAR szDesc[MAX_PATH];
 
-    if (GetUName)
-    {
         GetUName(wch, szDesc);
```
Comment (ThFabba) [inline base/applications/charmap/charmap.c]: Fix indentation of these please.

File: dll/win32/getuname/getuname.c [hunk 3]
```diff
@@ -18,14 +18,13 @@ GetUName(IN WORD wCharCode,
          OUT LPWSTR lpBuf)
 {
     WCHAR szDescription[256];
-    int res = LoadStringW(hInstance, wCharCode, szDescription, 256);
-    if (res != 0)
+    if (LoadStringW(hInstance, wCharCode, szDescription, 256))
```
Comment (ThFabba) [inline dll/win32/getuname/getuname.c]: Not an improvement.

**Comment by ThFabba** [comment]
> Based on the code from nls2txt,

Can you explain the relationship? I don't see anything in nls2txt that looks similar to the changes in this PR.
Thanks.

File: dll/win32/getuname/getuname.c (hunk 3 above)
Comment (peterooch) [reply dll/win32/getuname/getuname.c]: Isn't? LoadString returns 0(false) if no string got found for a specific code or if it pulls an empty string, and and a nonzero value(true) if there is an appropriate string in the resource table.

**Comment by peterooch** [comment]
nls2txt relationship:
[Cmake file](https://github.com/reactos/reactos/blob/c2c66aff7dacc62d125f2cd61d1167e9a2aa3fd6/modules/rosapps/applications/devutils/nls2txt/CMakeLists.txt)
[Function Prototype](https://github.com/reactos/reactos/blob/c2c66aff7dacc62d125f2cd61d1167e9a2aa3fd6/modules/rosapps/applications/devutils/nls2txt/precomp.h#L47)
[Being used](https://github.com/reactos/reactos/blob/c2c66aff7dacc62d125f2cd61d1167e9a2aa3fd6/modules/rosapps/applications/devutils/nls2txt/bestfit.c#L155)

File: dll/win32/getuname/getuname.c (hunk 3 above)
Comment (peterooch) [reply dll/win32/getuname/getuname.c]: No, you are right.
Should i revert the change?

Comment (ThFabba) [reply dll/win32/getuname/getuname.c]: Yes, please. It seems unnecessary.
Thanks.

Comment (peterooch) [reply dll/win32/getuname/getuname.c]: Done

**Comment by gedmurphy** [comment]
The code was originally written this way so charmap could be built outside of reactos. These changes now tie the code to the reactos tree. You can argue whether that's an improvement or a step backwards.

**Comment by learn-more** [comment]
@gedmurphy Since you are the author, why don't you make a case if this is an improvement or a step backwards? ;)

**Comment by Mouvedia** [comment]
@peterooch check https://jira.reactos.org/browse/CORE-10518?focusedCommentId=74889#comment-74889

**Comment by ThFabba** [comment]
How is that related to this PR?

**Pull request merged**